run_tests.sh

```diff
@@ -4,13 +4,25 @@ set -ex
 
 dir=$(pwd)
 # list of all modules to test
-modules=". client/cmd/dexcctl"
+modules=". server/cmd/dcrdex client/cmd/dexc client/cmd/dexcctl"
 
-# Test each module separately.
+# For each module, run go mod tidy, build and run test.
 for m in $modules
 do
 	cd $dir/$m
-	# run tests
+
+  # run `go mod tidy` and fail if the git status of go.mod and/or
+  # go.sum changes
+  MOD_STATUS=$(git status --porcelain go.mod go.sum)
+  go mod tidy
+  UPDATED_MOD_STATUS=$(git status --porcelain go.mod go.sum)
+  if [ "$UPDATED_MOD_STATUS" != "$MOD_STATUS" ]; then
+    echo "$m: running 'go mod tidy' modified go.mod and/or go.sum"
+    exit 1
+  fi
```

Comment:
Indentation is off, but I'm good with this.

Reply:
Pfft. I noticed the indentation issue when I ran `git diff` but didn't really make sense of it at the time. Missed the enclosing `for...do` block.

Reply:
I think we also got a preview of the error. https://github.com/decred/dcrdex/runs/777153624#step:5:86
Can we output git status and maybe even git diff when one of the two files were modified as in this failure?

Reply:
Ohh, I think we forgot something with the dcrjson change in your last PR.  We'll need to make the dcrjson/v3 require and import changes instead of the default one that go mod tidy adds.

Reply:
I'm thinking just `git diff` as that output will imply the git status of both files.

Reply:
Now displays the diff: https://github.com/decred/dcrdex/pull/492/checks?check_run_id=777371610#step:5:157.

Gonna fix the import issue next.

Reply:
Nice:

```
diff --git a/go.mod b/go.mod
index 41d651f..382839d 100644
--- a/go.mod
+++ b/go.mod
@@ -16,0 +17 @@ require (
+	github.com/decred/dcrd/dcrjson v1.2.0
diff --git a/go.sum b/go.sum
index 85b4aa5..5b40c82 100644
--- a/go.sum
+++ b/go.sum
@@ -64,0 +65,2 @@ github.com/decred/dcrd/dcrec/secp256k1/v2 v2.0.0/go.mod h1:3s92l0paYkZoIHuj4X93T
+github.com/decred/dcrd/dcrjson v1.2.0 h1:3BFFQHq3/YO/zae9WLxQkXsX6AXKx3+M8H3yk4oXZi0=
+github.com/decred/dcrd/dcrjson v1.2.0/go.mod h1:ozddIaeF+EAvZZvFuB3zpfxhyxBGfvbt22crQh+PYuI=
```